client/src/app/core/services/http.service.ts

```diff
@@ -18,30 +19,81 @@ export enum HTTPMethod {
 @Injectable({
     providedIn: 'root'
 })
-
 export class HttpService {
     /**
      * Construct a HttpService
      *
      * @param http The HTTP Client
      */
-    public constructor(private http: HttpClient) {}
+    public constructor(private http: HttpClient, private translate: TranslateService) {}
 
     private async send<T>(url: string, method: HTTPMethod, data?: any): Promise<T> {
         if (!url.endsWith('/')) {
             url += '/';
         }
 
         const options = {
-            body: data,
+            body: data
         };
 
         try {
             const response = await this.http.request<T>(method, url, options).toPromise();
             return response;
         } catch (e) {
-            console.log("error", e);
-            throw e;
+            throw this.handleError(e);
+        }
+    }
+
+    /**
+     * Takes an error thrown by the HttpClient. Processes it to return a string that can
+     * be presented to the user.
+     * @param e The error thrown.
+     */
+    private handleError(e: any): string {
+        let error = this.translate.instant('Error') + ': ';
+        // If the error is no HttpErrorResponse, it's not clear what is wrong.
+        if (!(e instanceof HttpErrorResponse)) {
+            console.error('Unknown error thrown by the http client: ', e);
+            error += this.translate.instant('An unknown error occurred.');
+            return error;
+        }
+
+        if (!e.error) {
+            error += this.translate.instant("The server didn't respond.");
+        } else if (typeof e.error === 'object') {
+            if (e.error.detail) {
+                error += this.processErrorTexts(e.error.detail);
+            } else {
+                error = Object.keys(e.error)
+                    .map(key => {
+                        const capitalizedKey = key.charAt(0).toUpperCase() + key.slice(1);
+                        return this.translate.instant(capitalizedKey) + ': ' + this.processErrorTexts(e.error[key]);
+                    })
+                    .join(', ');
+            }
+        } else if (e.status === 500) {
+            error += this.translate.instant('A server error occured. Please contact your system administrator.');
+        } else if (e.status > 500) {
+            error += this.translate
+                .instant('The server cound not be reached (%%code%%).')
```

Comment:
The TranslateParser understands parameters. Replacing this is not necessary.

from the documentation of ngx translate:

>instant(key: string|Array<string>, interpolateParams?: Object): string|Object: Gets the instant translated value of a key (or an array of keys). /!\ This method is synchronous and the default file loader is asynchronous. You are responsible for knowing when your translations have been loaded and it is safe to use this method. If you are not sure then you should use the get method instead.

Reply:
`this.translate.instant('The server cound not be reached ({{ code }}).', {code: e.status});` would be fine, but does not work :(
I'll create an issue for this and leave the replace there..